spec/unit/spark_api/request_spec.rb

```diff
@@ -248,7 +254,23 @@ def version()
           expect(number.to_s).to eq(BigDecimal.new("9999999999999999999999999.99").to_s)
         end
       end
-      
+
+      # This is a weird feature and it also gets a weird test that probably
+      # merits explanation:
+      #
+      # We have only stubbed POST for this route, so the below succeeding
+      # proves that we have converted our GET into a POST. That POST also is
+      # stubbed such that it proves that we are passing along the expected
+      # override_headers options, though they get converted to params on the URL
+      # by the mock connection along the way. Additionally, it demonstrates that
+      # we turn the params into a body which excludes the http_method_override
+      # param as well as the override_headers param. Another approach would to
+      # create a smarter stubbed connection that processes the options like
+      # SparkApi::Authentication::Oauth2 does but this did not seem worthwhile
+      # given the limited scope of this.
+      it "should should convert GET to POST if we http_method_override: true is supplied" do
+        expect(subject.get('/routetoproveweareposting', { http_method_override: true, some_param: "some_value" }).success?).to be true
```

Comment:
I think this test and the PR description itself are a good illustration of the low-level interface, but I'm curious how this would be leveraged through a model, e.g. via `Listing.find(:all, ...)`?

Reply:
Ah, sure, I should have shown that in action too. It's much the same.
```
[2] pry(main)> SparkApi::Models::Listing.find(:all, _select: (["ListPrice"] * 1000).join(','), http_method_override: true, _limit: 1)
/usr/local/rvm/gems/ruby-2.4.1/gems/faraday-0.9.2/lib/faraday/request/retry.rb:30: warning: constant ::Fixnum is deprecated
=> [#<SparkApi::Models::Listing:0x00000001632b70
  @attributes=
   {"Id"=>"20190304192247683399000000",
    "ResourceUri"=>"/v1/listings/20190304192247683399000000",
    "StandardFields"=>{"ListPrice"=>200000.0, "MlsId"=>"20181101193330369044000000"},
    "LastCachedTimestamp"=>nil,
    "DisplayCompliance"=>{"View"=>"Detail"}},
  @constraints=[],
  @documents=[],
  @errors=[],
  @open_houses=[],
  @photos=[],
  @rental_calendars=[],
  @tour_of_homes=[],
  @videos=[],
  @virtual_tours=[]>]
```